docs/bookstore/page-2/sentiment-analysis-service-for-bookstore-reviews.md

```diff
@@ -0,0 +1,414 @@
+
+    ```python
+    from parliament import Context
+    from flask import Request, request, jsonify
+    import json
+    from textblob import TextBlob
+    from time import sleep
+    from cloudevents.http import CloudEvent, to_structured
+
+    # The function to convert the sentiment analysis result into a CloudEvent
+    def create_cloud_event(inputText, badWordResult, data):
+        attributes = {
+            "type": "moderated-comment",
+            "source": "sentiment-analysis",
+            "datacontenttype": "application/json",
+            "sentimentResult": data,
+            "badwordfilter": badWordResult,
+        }
+
+        # Put the sentiment analysis result into a dictionary
+        data = {
+            "reviewText": inputText,
+            "badWordResult": badWordResult,
+            "sentimentResult": data,
+        }
+
+        # Create a CloudEvent object
+        event = CloudEvent(attributes, data)
+        return event
+
+    def analyze_sentiment(text):
+        analysis = TextBlob(text["reviewText"])
+        sentiment = "neutral"
+
+        if analysis.sentiment.polarity > 0:
+            sentiment = "positive"
+        elif analysis.sentiment.polarity < 0:
+            sentiment = "negative"
+
+        badWordResult = ""
+        try:
+            badWordResult = text["badWordResult"]
+        except:
+            pass
+
+        # Convert the sentiment into a CloudEvent
+        sentiment = create_cloud_event(text["reviewText"], badWordResult, sentiment)
+        return sentiment
+
+    def main(context: Context):
+        """
+        Function template
+        The context parameter contains the Flask request object and any
+        CloudEvent received with the request.
+        """
+
+        print("Sentiment Analysis Received CloudEvent: ", context.cloud_event)
+
+        # Add your business logic here
+        return analyze_sentiment(context.cloud_event.data)
+    ```
+
+### **Step 3: Configure the dependencies**
+
+![Image9](images/image9.png)
+
+The `requirements.txt` file contains the dependencies for the function. You can add the following dependencies to the `requirements.txt` file:
+???+ abstract "_sentiment-analysis-app/requirements.txt_"
+
+    ```
+    Flask==3.0.2
+    textblob==0.18.0.post0
+    parliament-functions==0.1.0
+    cloudevents==1.10.1
+    ```
+
+Knative Function will automatically install the dependencies listed here when you build the function.
+
+### **Step 4: Configure the pre-built environment**
+
+![Image11](images/image11.png)
+
+In order to properly use the `textblob` library, you need to download the corpora, which is a large collection of text data that is used to train the sentiment analysis model. You can do this by creating a new file called `setup.py`, Knative Function will ensure that the `setup.py` file is executed after the dependencies have been installed.
+
+The `setup.py` file should contain the following code for your bookstore:
```

Comment:
I am not sure this is all really needed.  

I think the tutorial is already complex, as it covers a lot of things.   

For folks have no idea on python, I guess they are all lost .

Reply:
In order to use textblob library, we have to download their corpora file before hand. Unless there are any other methods we can do that, I think the `setup.py` part tutorial is necessary, and this approach is not documented anywhere in Knative's website. WDYT? @matzew